### PR TITLE
Cleaning up a couple bugs, adding configuration for search heads, and instance type updates

### PR DESCRIPTION
**Pull request by billbartlett**
*Issue #, if available:*

*Description of changes:*

- init.d script fixes.
- instance type updates and updated defaults
- larger minimum EBS volume requirements due to AMI construction, (both indexers and search heads)
- search head disk parameter added
- indexer cluster search factor parameter added, and limited from 2-4
- changed default replication factor to 2, and updated range to 2-4 from 2-5
- changed instance type of deployer and cluster master to c5 family

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by billbartlett** [comment]
I believe I've fixed the conflicts.  Please have a look and let me know if you see anything outstanding.

File: templates/splunk-enterprise-master.template [hunk 1]
```diff
@@ -235,7 +256,9 @@
                         "SplunkLicensePath",
                         "SplunkIndexerCount",
                         "SplunkIndexerDiskSize",
+                        "SplunkSearchHeadDiskSize",
```
Comment (sshvans) [inline templates/splunk-enterprise-master.template]: Parameter label is missing for this parameter.

File: templates/splunk-enterprise.template [hunk 2]
```diff
@@ -91,6 +96,16 @@
             "Type": "AWS::EC2::Subnet::Id",
             "Default": ""
         },
+        "QSS3BucketName": {
```
Comment (sshvans) [inline templates/splunk-enterprise.template]: The bucket name default value should be aws-quickstart

**Comment by billbartlett** [comment]
Since this PR, I have made additional changes.  I will close this one, and make a new one that addresses your suggestions here.

**Pull request closed without merging**